SyncPlan test timeout on waiting for the sync task to get created

This is happening spontaneously, most probably because of the server load.

the last time it caused these two to fail:
`test_positive_synchronize_rh_product_future_sync_date`
`test_positive_synchronize_rh_product_past_sync_date`

- i'll try to dig out the whole tier4 logs to see, whether i can find the list of the tasks.

On upgrade box, i see these failures, 

 ```
tests.foreman.cli.test_syncplan.SyncPlanTestCase.test_positive_synchronize_custom_product_daily_recurrence
tests.foreman.cli.test_syncplan.SyncPlanTestCase.test_positive_synchronize_custom_product_past_sync_date
tests.foreman.api.test_syncplan.SyncPlanSynchronizeTestCase.test_positive_synchronize_rh_product_future_sync_date
```

Error:
`AssertionError: No task was found using query 'resource_type = Katello::Repository and owner.login = foreman_admin and resource_id = 636'`

When I fire up the same query now, I can see the task was created and completed successfully.

--- a/robottelo/api/utils.py
+++ b/robottelo/api/utils.py
@@ -5,6 +5,7 @@ import time
 from fauxfactory import gen_string
 from inflector import Inflector
 from nailgun import entities, entity_mixins
+from nailgun.client import request
 from robottelo import ssh
 from robottelo.config import settings
 from robottelo.config.base import ImproperlyConfigured
@@ -622,3 +623,70 @@ def wait_for_tasks(search_query, search_rate=1, max_tries=10, poll_rate=None,
         raise AssertionError(
             "No task was found using query '{}'".format(search_query))
     return tasks
+
+
+def wait_for_syncplan_tasks(repo_backend_id=None, timeout=10, repo_name=None):
+    """Search the pulp tasks and identify repositories sync tasks with
+    specified name or backend_identifier
+
+    :param repo_backend_id: The Backend ID for the repository to identify the
+        repo in Pulp environment
+    :param timeout: Value to decided how long to check for the Sync task
+    :param repo_name: If repo_backend_id can not be passed, pass the repo_name
+    """
+    if repo_name:
+            repo_backend_id = entities.Repository().search(query={
+                        'search': 'name="{0}"'.format(repo_name),
+                        'per_page': 1000,
+                    })[0].backend_identifier
+    # Fetch the Pulp password
+    pulp_pass = ssh.command(
+        'grep "^default_password" /etc/pulp/server.conf |'
+        ' awk \'{print $2}\''
+    ).stdout[0]
+    # Set the Timeout value
+    timeup = time.time() + int(timeout) * 60
+    # Search Filter to filter out the task based on backend-id and sync action
+    filtered_req = {
+        'criteria': {
+            'filters': {
+                'tags': {
+                    '$in': [
+                        "pulp:repository:{0}".format(repo_backend_id)]
+                },
+                'task_type': {
+                    '$in': ["pulp.server.managers.repo.sync.sync"]
+                }
+            }
+        }
+    }
+    while True:
+        if time.time() > timeup:
+            raise entities.APIResponseError(
+                'Pulp task with repo_id {0} not found'.format(repo_backend_id)
+            )
+        # Send request to pulp API to get the task info
+        req = request('POST',
+                      '{0}/pulp/api/v2/tasks/search/'.format(
+                          settings.server.get_url()),
+                      verify=False,
+                      auth=('admin', '{0}'.format(pulp_pass)),
+                      headers={'content-type': 'application/json'},
+                      data=filtered_req)
+        # Check Status code of response
+        if req.status_code != 200:
+            raise entities.APIResponseError(
+                'Pulp task with repo_id {0} not found'.format(repo_backend_id))
+        # Check content of response
+        # It is '[]' string for empty content when backend_identifier is wrong
+        if len(req.content) > 2:
+            if req.json()[0].get('state') in ['finished']:
+                return True
+            elif req.json()[0].get('error'):
+                raise AssertionError(
+                    "Pulp task with repo_id {0} errored or not "
+                    "found: '{1}'".format(repo_backend_id,
+                                          req.json().get('error')
+                                          )
+                )
+        time.sleep(2)

--- a/tests/foreman/api/test_syncplan.py
+++ b/tests/foreman/api/test_syncplan.py
@@ -23,7 +23,11 @@ from fauxfactory import gen_string
 from nailgun import client, entities
 from random import sample
 from robottelo import manifests
-from robottelo.api.utils import enable_rhrepo_and_fetchid, wait_for_tasks
+from robottelo.api.utils import (
+    enable_rhrepo_and_fetchid,
+    wait_for_tasks,
+    wait_for_syncplan_tasks
+)
 from robottelo.config import settings
 from robottelo.constants import PRDS, REPOS, REPOSET
 from robottelo.datafactory import (
@@ -581,7 +585,16 @@ class SyncPlanSynchronizeTestCase(APITestCase):
         cls.org = entities.Organization().create()
 
     @staticmethod
-    def validate_task_status(repo_id, max_tries=10):
+    def validate_task_status(repo_id, max_tries=10, repo_backend_id=None):
+        """Wait for Pulp and foreman_tasks to complete or timeout
+
+        :param repo_id: Repository Id to identify the correct task
+        :param max_tries: Max tries to poll for the task creation
+        :param repo_backend_id: Backend identifier of repository to filter the
+            pulp tasks
+        """
+        if repo_backend_id:
+            wait_for_syncplan_tasks(repo_backend_id)
         wait_for_tasks(
             search_query='resource_type = Katello::Repository'
                          ' and owner.login = foreman_admin'
@@ -630,11 +643,6 @@ class SyncPlanSynchronizeTestCase(APITestCase):
 
         :CaseLevel: System
         """
-        sync_plan = entities.SyncPlan(
-            organization=self.org,
-            enabled=True,
-            sync_date=datetime.utcnow(),
-        ).create()
         product = entities.Product(organization=self.org).create()
         repo = entities.Repository(product=product).create()
         # Verify product is not synced and doesn't have any content
@@ -642,7 +650,12 @@ class SyncPlanSynchronizeTestCase(APITestCase):
             self.validate_task_status(repo.id, max_tries=2)
         self.validate_repo_content(
             repo, ['erratum', 'package', 'package_group'], after_sync=False)
-        # Associate sync plan with product
+        # Create and Associate sync plan with product
+        sync_plan = entities.SyncPlan(
+            organization=self.org,
+            enabled=True,
+            sync_date=datetime.utcnow(),
+        ).create()
         sync_plan.add_products(data={'product_ids': [product.id]})
         # Verify product was not synced right after it was added to sync plan
         with self.assertRaises(AssertionError):
@@ -666,15 +679,15 @@ class SyncPlanSynchronizeTestCase(APITestCase):
         """
         interval = 60 * 60  # 'hourly' sync interval in seconds
         delay = 4 * 60
+        product = entities.Product(organization=self.org).create()
+        repo = entities.Repository(product=product).create()
+        # Create and Associate sync plan with product
         sync_plan = entities.SyncPlan(
             organization=self.org,
             enabled=True,
             interval=u'hourly',
             sync_date=datetime.utcnow() - timedelta(seconds=interval - delay),
         ).create()
-        product = entities.Product(organization=self.org).create()
-        repo = entities.Repository(product=product).create()
-        # Associate sync plan with product
         sync_plan.add_products(data={'product_ids': [product.id]})
         # Verify product is not synced and doesn't have any content
         self.logger.info('Waiting {0} seconds to check product {1}'
@@ -688,8 +701,13 @@ class SyncPlanSynchronizeTestCase(APITestCase):
         self.logger.info('Waiting {0} seconds to check product {1}'
                          ' was synced'.format(delay, product.name))
         sleep(delay * 3/4)
+        # Update with the current UTC time
+        sync_plan.sync_date = datetime.utcnow() + timedelta(seconds=delay)
+        sync_plan.update(['sync_date'])
         # Verify product was synced successfully
-        self.validate_task_status(repo.id)
+        self.validate_task_status(repo.id,
+                                  repo_backend_id=repo.backend_identifier
+                                  )
         self.validate_repo_content(
             repo, ['erratum', 'package', 'package_group'])
 
@@ -705,11 +723,6 @@ class SyncPlanSynchronizeTestCase(APITestCase):
         :CaseLevel: System
         """
         delay = 4 * 60  # delay for sync date in seconds
-        sync_plan = entities.SyncPlan(
-            organization=self.org,
-            enabled=True,
-            sync_date=datetime.utcnow() + timedelta(seconds=delay),
-        ).create()
         product = entities.Product(organization=self.org).create()
         repo = entities.Repository(product=product).create()
         # Verify product is not synced and doesn't have any content
@@ -717,7 +730,12 @@ class SyncPlanSynchronizeTestCase(APITestCase):
             self.validate_task_status(repo.id, max_tries=2)
         self.validate_repo_content(
             repo, ['erratum', 'package', 'package_group'], after_sync=False)
-        # Associate sync plan with product
+        # Create and Associate sync plan with product
+        sync_plan = entities.SyncPlan(
+            organization=self.org,
+            enabled=True,
+            sync_date=datetime.utcnow() + timedelta(seconds=delay),
+        ).create()
         sync_plan.add_products(data={'product_ids': [product.id]})
         # Wait half of expected time
         self.logger.info('Waiting {0} seconds to check product {1}'
@@ -732,8 +750,13 @@ class SyncPlanSynchronizeTestCase(APITestCase):
         self.logger.info('Waiting {0} seconds to check product {1}'
                          ' was synced'.format(delay, product.name))
         sleep(delay * 3/4)
+        # Update with the current UTC time
+        sync_plan.sync_date = datetime.utcnow() + timedelta(seconds=delay)
+        sync_plan.update(['sync_date'])
         # Verify product was synced successfully
-        self.validate_task_status(repo.id)
+        self.validate_task_status(repo.id,
+                                  repo_backend_id=repo.backend_identifier
+                                  )
         self.validate_repo_content(
             repo, ['erratum', 'package', 'package_group'])
 
@@ -750,11 +773,6 @@ class SyncPlanSynchronizeTestCase(APITestCase):
         :CaseLevel: System
         """
         delay = 6 * 60  # delay for sync date in seconds
-        sync_plan = entities.SyncPlan(
-            organization=self.org,
-            enabled=True,
-            sync_date=datetime.utcnow() + timedelta(seconds=delay),
-        ).create()
         products = [
             entities.Product(organization=self.org).create()
             for _ in range(3)
@@ -768,7 +786,12 @@ class SyncPlanSynchronizeTestCase(APITestCase):
         for repo in repos:
             with self.assertRaises(AssertionError):
                 self.validate_task_status(repo.id)
-        # Associate sync plan with products
+        # Create and Associate sync plan with products
+        sync_plan = entities.SyncPlan(
+            organization=self.org,
+            enabled=True,
+            sync_date=datetime.utcnow() + timedelta(seconds=delay),
+        ).create()
         sync_plan.add_products(data={
             'product_ids': [product.id for product in products]})
         # Wait half of expected time
@@ -783,9 +806,14 @@ class SyncPlanSynchronizeTestCase(APITestCase):
         self.logger.info('Waiting {0} seconds to check products'
                          ' were synced'.format(delay))
         sleep(delay * 3/4)
+        # Update with the current UTC time
+        sync_plan.sync_date = datetime.utcnow() + timedelta(seconds=delay)
+        sync_plan.update(['sync_date'])
         # Verify product was synced successfully
         for repo in repos:
-            self.validate_task_status(repo.id)
+            self.validate_task_status(repo.id,
+                                      repo_backend_id=repo.backend_identifier
+                                      )
             self.validate_repo_content(
                 repo, ['erratum', 'package', 'package_group'])
 
@@ -845,8 +873,14 @@ class SyncPlanSynchronizeTestCase(APITestCase):
         self.logger.info('Waiting {0} seconds to check product {1}'
                          ' was synced'.format(delay, product.name))
         sleep(delay * 3/4)
+        # Update with the current UTC time
+        sync_plan.sync_date = datetime.utcnow() - timedelta(
+            seconds=interval - delay)
+        sync_plan.update(['sync_date'])
         # Verify product was synced successfully
-        self.validate_task_status(repo.id)
+        self.validate_task_status(repo.id,
+                                  repo_backend_id=repo.backend_identifier
+                                  )
         self.validate_repo_content(
             repo, ['erratum', 'package', 'package_group'])
 
@@ -870,12 +904,6 @@ class SyncPlanSynchronizeTestCase(APITestCase):
                 data={'organization_id': org.id},
                 files={'content': manifest.content},
             )
-        sync_plan = entities.SyncPlan(
-            organization=org,
-            enabled=True,
-            interval=u'hourly',
-            sync_date=datetime.utcnow() + timedelta(seconds=delay),
-        ).create()
         repo_id = enable_rhrepo_and_fetchid(
             basearch='x86_64',
             org_id=org.id,
@@ -889,7 +917,13 @@ class SyncPlanSynchronizeTestCase(APITestCase):
             organization=org,
         ).search()[0]
         repo = entities.Repository(id=repo_id).read()
-        # Associate sync plan with product
+        sync_plan = entities.SyncPlan(
+            organization=org,
+            enabled=True,
+            interval=u'hourly',
+            sync_date=datetime.utcnow() + timedelta(seconds=delay),
+        ).create()
+        # Create and Associate sync plan with product
         sync_plan.add_products(data={'product_ids': [product.id]})
         # Verify product is not synced and doesn't have any content
         with self.assertRaises(AssertionError):
@@ -909,8 +943,13 @@ class SyncPlanSynchronizeTestCase(APITestCase):
         self.logger.info('Waiting {0} seconds to check product {1}'
                          ' was synced'.format(delay, product.name))
         sleep(delay * 3/4)
+        # Update with the current UTC time
+        sync_plan.sync_date = datetime.utcnow() + timedelta(seconds=delay)
+        sync_plan.update(['sync_date'])
         # Verify product was synced successfully
-        self.validate_task_status(repo.id)
+        self.validate_task_status(repo.id,
+                                  repo_backend_id=repo.backend_identifier
+                                  )
         self.validate_repo_content(
             repo, ['erratum', 'package', 'package_group'])
 
@@ -929,15 +968,15 @@ class SyncPlanSynchronizeTestCase(APITestCase):
         delay = 4 * 60
         start_date = datetime.utcnow() - timedelta(days=1)\
             + timedelta(seconds=delay)
+        product = entities.Product(organization=self.org).create()
+        repo = entities.Repository(product=product).create()
+        # Create and Associate sync plan with product
         sync_plan = entities.SyncPlan(
             organization=self.org,
             enabled=True,
             interval=u'hourly',
             sync_date=start_date,
         ).create()
-        product = entities.Product(organization=self.org).create()
-        repo = entities.Repository(product=product).create()
-        # Associate sync plan with product
         sync_plan.add_products(data={'product_ids': [product.id]})
         # Verify product is not synced and doesn't have any content
         self.logger.info('Waiting {0} seconds to check product {1}'
@@ -951,8 +990,15 @@ class SyncPlanSynchronizeTestCase(APITestCase):
         self.logger.info('Waiting {0} seconds to check product {1}'
                          ' was synced'.format(delay, product.name))
         sleep(delay * 3/4)
+        # Re-calculate and Update with the current UTC time
+        start_date = datetime.utcnow() - timedelta(days=1)\
+            + timedelta(seconds=delay)
+        sync_plan.sync_date = start_date
+        sync_plan.update(['sync_date'])
         # Verify product was synced successfully
-        self.validate_task_status(repo.id)
+        self.validate_task_status(repo.id,
+                                  repo_backend_id=repo.backend_identifier
+                                  )
         self.validate_repo_content(
             repo, ['erratum', 'package', 'package_group'])
 
@@ -974,15 +1020,15 @@ class SyncPlanSynchronizeTestCase(APITestCase):
         delay = 4 * 60
         start_date = datetime.utcnow() - timedelta(weeks=1) \
             + timedelta(seconds=delay)
+        product = entities.Product(organization=self.org).create()
+        repo = entities.Repository(product=product).create()
+        # Create and Associate sync plan with product
         sync_plan = entities.SyncPlan(
             organization=self.org,
             enabled=True,
             interval=u'weekly',
             sync_date=start_date,
         ).create()
-        product = entities.Product(organization=self.org).create()
-        repo = entities.Repository(product=product).create()
-        # Associate sync plan with product
         sync_plan.add_products(data={'product_ids': [product.id]})
         # Verify product is not synced and doesn't have any content
         self.logger.info('Waiting {0} seconds to check product {1}'
@@ -996,8 +1042,15 @@ class SyncPlanSynchronizeTestCase(APITestCase):
         self.logger.info('Waiting {0} seconds to check product {1}'
                          ' was synced'.format(delay, product.name))
         sleep(delay * 3/4)
+        # Re-calculate and Update with the current UTC time
+        start_date = datetime.utcnow() - timedelta(weeks=1) \
+            + timedelta(seconds=delay)
+        sync_plan.sync_date = start_date
+        sync_plan.update(['sync_date'])
         # Verify product was synced successfully
-        self.validate_task_status(repo.id)
+        self.validate_task_status(repo.id,
+                                  repo_backend_id=repo.backend_identifier
+                                  )
         self.validate_repo_content(
             repo, ['erratum', 'package', 'package_group'])
 

--- a/tests/foreman/cli/test_syncplan.py
+++ b/tests/foreman/cli/test_syncplan.py
@@ -19,7 +19,7 @@
 from datetime import datetime, timedelta
 from fauxfactory import gen_string
 from robottelo import manifests
-from robottelo.api.utils import wait_for_tasks
+from robottelo.api.utils import wait_for_tasks, wait_for_syncplan_tasks
 from robottelo.cli.base import CLIReturnCodeError
 from robottelo.cli.factory import (
     CLIFactoryError,
@@ -143,7 +143,16 @@ class SyncPlanTestCase(CLITestCase):
         return make_sync_plan(options)
 
     @staticmethod
-    def validate_task_status(repo_id, max_tries=10):
+    def validate_task_status(repo_id, max_tries=10, repo_name=None):
+        """Wait for Pulp and foreman_tasks to complete or timeout
+
+        :param repo_id: Repository Id to identify the correct task
+        :param max_tries: Max tries to poll for the task creation
+        :param repo_name: Repository name of repository to filter the
+            pulp tasks
+        """
+        if repo_name:
+            wait_for_syncplan_tasks(repo_name=repo_name)
         wait_for_tasks(
             search_query='resource_type = Katello::Repository'
                          ' and owner.login = foreman_admin'
@@ -474,8 +483,15 @@ class SyncPlanTestCase(CLITestCase):
         self.logger.info('Waiting {0} seconds to check product {1}'
                          ' was synced'.format(delay, product['name']))
         sleep(delay * 3/4)
+        # Re-calculate and Update with the current UTC time
+        SyncPlan.update({
+            u'id': sync_plan['id'],
+            u'sync-date': (
+              datetime.utcnow() - timedelta(seconds=interval - delay)
+            ).strftime("%Y-%m-%d %H:%M:%S"),
+        })
         # Verify product was synced successfully
-        self.validate_task_status(repo['id'])
+        self.validate_task_status(repo['id'], repo_name=repo['name'])
         self.validate_repo_content(
             repo, ['errata', 'package-groups', 'packages'])
 
@@ -521,8 +537,14 @@ class SyncPlanTestCase(CLITestCase):
         self.logger.info('Waiting {0} seconds to check product {1}'
                          ' was synced'.format(delay/2, product['name']))
         sleep(delay * 3/4)
+        # Re-calculate and Update with the current UTC time
+        SyncPlan.update({
+            u'id': sync_plan['id'],
+            u'sync-date': (datetime.utcnow() + timedelta(seconds=delay))
+                .strftime("%Y-%m-%d %H:%M:%S"),
+        })
         # Verify product was synced successfully
-        self.validate_task_status(repo['id'])
+        self.validate_task_status(repo['id'], repo_name=repo['name'])
         self.validate_repo_content(
             repo, ['errata', 'package-groups', 'packages'])
 
@@ -576,9 +598,15 @@ class SyncPlanTestCase(CLITestCase):
         self.logger.info('Waiting {0} seconds to check products'
                          ' were synced'.format(delay/2))
         sleep(delay * 3/4)
+        # Re-calculate and Update with the current UTC time
+        SyncPlan.update({
+            u'id': sync_plan['id'],
+            u'sync-date': (datetime.utcnow() + timedelta(seconds=delay))
+                .strftime("%Y-%m-%d %H:%M:%S"),
+        })
         # Verify product was synced successfully
         for repo in repos:
-            self.validate_task_status(repo['id'])
+            self.validate_task_status(repo['id'], repo_name=repo['name'])
             self.validate_repo_content(
                 repo, ['errata', 'package-groups', 'packages'])
 
@@ -648,8 +676,15 @@ class SyncPlanTestCase(CLITestCase):
         self.logger.info('Waiting {0} seconds to check product {1}'
                          ' was synced'.format(delay, product['name']))
         sleep(delay * 3/4)
+        # Re-calculate and Update with the current UTC time
+        SyncPlan.update({
+            u'id': sync_plan['id'],
+            u'sync-date': (
+              datetime.utcnow() - timedelta(seconds=interval - delay)
+            ).strftime("%Y-%m-%d %H:%M:%S"),
+        })
         # Verify product was synced successfully
-        self.validate_task_status(repo['id'])
+        self.validate_task_status(repo['id'], repo_name=repo['name'])
         self.validate_repo_content(repo, ['errata', 'packages'])
 
     @run_in_one_thread
@@ -718,8 +753,14 @@ class SyncPlanTestCase(CLITestCase):
         self.logger.info('Waiting {0} seconds to check product {1}'
                          ' was synced'.format(delay/2, product['name']))
         sleep(delay * 3/4)
+        # Re-calculate and Update with the current UTC time
+        SyncPlan.update({
+            u'id': sync_plan['id'],
+            u'sync-date': (datetime.utcnow() + timedelta(seconds=delay))
+                .strftime("%Y-%m-%d %H:%M:%S"),
+        })
         # Verify product was synced successfully
-        self.validate_task_status(repo['id'])
+        self.validate_task_status(repo['id'], repo_name=repo['name'])
         self.validate_repo_content(repo, ['errata', 'packages'])
 
     @tier3
@@ -763,8 +804,15 @@ class SyncPlanTestCase(CLITestCase):
         self.logger.info('Waiting {0} seconds to check product {1}'
                          ' was synced'.format(delay, product['name']))
         sleep(delay * 3/4)
+        # Re-calculate and Update with the current UTC time
+        start_date = datetime.utcnow() - timedelta(days=1)\
+            + timedelta(seconds=delay)
+        SyncPlan.update({
+            u'id': sync_plan['id'],
+            u'sync-date': start_date.strftime("%Y-%m-%d %H:%M:%S"),
+        })
         # Verify product was synced successfully
-        self.validate_task_status(repo['id'])
+        self.validate_task_status(repo['id'], repo_name=repo['name'])
         self.validate_repo_content(
             repo, ['errata', 'package-groups', 'packages'])
 
@@ -810,7 +858,14 @@ class SyncPlanTestCase(CLITestCase):
         self.logger.info('Waiting {0} seconds to check product {1}'
                          ' was synced'.format(delay, product['name']))
         sleep(delay * 3/4)
+        # Re-calculate and Update with the current UTC time
+        start_date = datetime.utcnow() - timedelta(weeks=1)\
+            + timedelta(seconds=delay)
+        SyncPlan.update({
+            u'id': sync_plan['id'],
+            u'sync-date': start_date.strftime("%Y-%m-%d %H:%M:%S"),
+        })
         # Verify product was synced successfully
-        self.validate_task_status(repo['id'])
+        self.validate_task_status(repo['id'], repo_name=repo['name'])
         self.validate_repo_content(
             repo, ['errata', 'package-groups', 'packages'])

--- a/tests/foreman/ui/test_syncplan.py
+++ b/tests/foreman/ui/test_syncplan.py
@@ -20,7 +20,11 @@ from fauxfactory import gen_string
 from nailgun import entities
 from random import choice
 from robottelo import manifests
-from robottelo.api.utils import enable_rhrepo_and_fetchid, wait_for_tasks
+from robottelo.api.utils import (
+    enable_rhrepo_and_fetchid,
+    wait_for_tasks,
+    wait_for_syncplan_tasks
+)
 from robottelo.constants import PRDS, REPOS, REPOSET, SYNC_INTERVAL
 from robottelo.datafactory import (
     filtered_datapoint,
@@ -64,7 +68,16 @@ class SyncPlanTestCase(UITestCase):
         cls.organization = entities.Organization().create()
 
     @staticmethod
-    def validate_task_status(repo_id, max_tries=10):
+    def validate_task_status(repo_id, max_tries=10, repo_backend_id=None):
+        """Wait for Pulp and foreman_tasks to complete or timeout
+
+        :param repo_id: Repository Id to identify the correct task
+        :param max_tries: Max tries to poll for the task creation
+        :param repo_backend_id: Backend identifier of repository to filter the
+            pulp tasks
+        """
+        if repo_backend_id:
+            wait_for_syncplan_tasks(repo_backend_id)
         wait_for_tasks(
             search_query='resource_type = Katello::Repository'
                          ' and owner.login = foreman_admin'
@@ -605,7 +618,9 @@ class SyncPlanTestCase(UITestCase):
                              ' was synced'.format(delay, product.name))
             sleep(delay * 3/4)
             # Verify product was synced successfully
-            self.validate_task_status(repo.id)
+            self.validate_task_status(repo.id,
+                                      repo_backend_id=repo.backend_identifier
+                                      )
             self.validate_repo_content(
                 repo,
                 ['erratum', 'package', 'package_group'],
@@ -664,7 +679,9 @@ class SyncPlanTestCase(UITestCase):
                              ' was synced'.format(delay/2, product.name))
             sleep(delay * 3/4)
             # Verify product was synced successfully
-            self.validate_task_status(repo.id)
+            self.validate_task_status(repo.id,
+                                      repo_backend_id=repo.backend_identifier
+                                      )
             self.validate_repo_content(
                 repo,
                 ['erratum', 'package', 'package_group'],
@@ -725,7 +742,9 @@ class SyncPlanTestCase(UITestCase):
                              ' were synced'.format(delay*2/3))
             sleep(delay * 3/4)
             # Verify product was synced successfully
-            self.validate_task_status(repo.id)
+            self.validate_task_status(repo.id,
+                                      repo_backend_id=repo.backend_identifier
+                                      )
             for repo in repos:
                 self.validate_repo_content(
                     repo,
@@ -799,7 +818,9 @@ class SyncPlanTestCase(UITestCase):
                              ' was synced'.format(delay, PRDS['rhel']))
             sleep(delay * 3/4)
             # Verify product was synced successfully
-            self.validate_task_status(repo.id)
+            self.validate_task_status(repo.id,
+                                      repo_backend_id=repo.backend_identifier
+                                      )
             self.validate_repo_content(
                 repo,
                 ['erratum', 'package', 'package_group'],
@@ -807,6 +828,7 @@ class SyncPlanTestCase(UITestCase):
 
     @run_in_one_thread
     @tier4
+    @upgrade
     def test_positive_synchronize_rh_product_future_sync_date(self):
         """Create a sync plan with sync date in a future and sync one RH
         product with it automatically.
@@ -866,7 +888,9 @@ class SyncPlanTestCase(UITestCase):
                              ' was synced'.format(delay/2, PRDS['rhel']))
             sleep(delay * 3/4)
             # Verify product was synced successfully
-            self.validate_task_status(repo.id)
+            self.validate_task_status(repo.id,
+                                      repo_backend_id=repo.backend_identifier
+                                      )
             self.validate_repo_content(
                 repo,
                 ['erratum', 'package', 'package_group'],
@@ -920,7 +944,9 @@ class SyncPlanTestCase(UITestCase):
                              ' was synced'.format(delay, product.name))
             sleep(delay * 3/4)
             # Verify product was synced successfully
-            self.validate_task_status(repo.id)
+            self.validate_task_status(repo.id,
+                                      repo_backend_id=repo.backend_identifier
+                                      )
             self.validate_repo_content(
                 repo,
                 ['erratum', 'package', 'package_group'],
@@ -929,7 +955,6 @@ class SyncPlanTestCase(UITestCase):
     @skip_if_bug_open('bugzilla', '1396647')
     @skip_if_bug_open('bugzilla', '1460146')
     @tier3
-    @upgrade
     def test_positive_synchronize_custom_product_weekly_recurrence(self):
         """Create a daily sync plan with past datetime as a sync date,
         add a custom product and verify the product gets synchronized
@@ -979,7 +1004,9 @@ class SyncPlanTestCase(UITestCase):
                              ' was synced'.format(delay, product.name))
             sleep(delay * 3/4)
             # Verify product was synced successfully
-            self.validate_task_status(repo.id)
+            self.validate_task_status(repo.id,
+                                      repo_backend_id=repo.backend_identifier
+                                      )
             self.validate_repo_content(
                 repo,
                 ['erratum', 'package', 'package_group'],

